src/UserProfileService/Controllers/V2/UsersController.cs

```diff
@@ -208,12 +205,12 @@ await _readService.GetProfilesAsync<UserBasic, GroupBasic, OrganizationBasic>(
     public async Task<IActionResult> GetOwnProfileAsync(CancellationToken cancellationToken = default)
     {
         _logger.EnterMethod();
-        string currentUserId = await _userContextStore.GetIdOfCurrentUserAsync();
 
-        var profile =
-            await _readService.GetProfileAsync<User>(
+        string currentUserId = _userContextStore.GetIdOfCurrentUser();
+
+        List<IProfile> profile =
+            await _readService.GetProfileByExternalOrInternalIdAsync<User, Group, Organization>(
                 currentUserId,
-                RequestedProfileKind.User,
                 cancellationToken: cancellationToken);
 
         IActionResult result =
```

Comment:
You changed the type of `profile`. Before your modification it was a single object - not it is a collectable/list. You should take one appropriate profile instead of all possible (even if the list just contains of one single item).
Like 
```csharp
var profiles = await _readService.Get...
var profile = profiles?.FirstOrDefault(p is User);

if (profile == null)
   return NotFound() / Error;
```

Reply:
Also done.